zp-relayer/direct-deposit/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { logger } from '@/services/appLogger'
+import { redis } from '@/services/redisClient'
+import config from '@/configs/baseConfig'
+import type { DirectDeposit } from '@/queue/poolTxQueue'
+
+const serviceKey = 'direct-deposit'
+const lastBlockRedisKey = `${serviceKey}:lastProcessedBlock`
+
+export let lastProcessedBlock = Math.max(config.startBlock - 1, 0)
+
+export async function getLastProcessedBlock() {
+  const result = await redis.get(lastBlockRedisKey)
+  logger.debug('Last Processed block obtained', { fromRedis: result, fromConfig: lastProcessedBlock })
+  lastProcessedBlock = result ? parseInt(result, 10) : lastProcessedBlock
+}
+
+export function updateLastProcessedBlock(lastBlockNumber: number) {
+  lastProcessedBlock = lastBlockNumber
+  return redis.set(lastBlockRedisKey, lastProcessedBlock)
+}
+
+export function parseDirectDepositEvent(o: Record<string, any>): DirectDeposit {
```

Comment:
Why not `o as DirectDeposit`?

Reply:
Here, we return a new object, so explicit type is needed.
`getEvents` method returns object with some redundant field, e.g. for event `E(uint64 a, uint64 b)` it returns:
```
{
    "a": 123,
    "b": 456,
    "0": 123,
    "1": 456
}
```
So, in this method I remove this integer duplicated indices.